pkg/kubelet/server/remotecommand/exec.go

```diff
@@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
 
 	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
 	if err != nil {
-		msg := fmt.Sprintf("error executing command in container: %v", err)
-		runtime.HandleError(errors.New(msg))
-		fmt.Fprint(ctx.errorStream, msg)
+		m := ExitError{}
```

Comment:
Yeah, looking at the code the way it's currently structured, it's not
immediately easy to send back different responses based on protocol version
right here. I guess we could stick the protocol on the context object as
part of createStreams.

On Thu, Aug 4, 2016 at 8:28 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In pkg/kubelet/server/remotecommand/exec.go
> <https://github.com/kubernetes/kubernetes/pull/26541#discussion_r73511874>
> :
>
> > @@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
> >
> >  	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
> >  	if err != nil {
> > -		msg := fmt.Sprintf("error executing command in container: %v", err)
> > -		runtime.HandleError(errors.New(msg))
> > -		fmt.Fprint(ctx.errorStream, msg)
> > +		m := ExitError{}
>
> That was my initial rational as well. But adding v4 is cleaner and not
> much more work. We have to find a way to send different error messages
> anyway, depending on the version.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26541/files/0a10e2dcdaf86a90b5a718e96d44cc9e5286ca06#r73511874>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYgdC0PlazH89hhQjAeVBAHJREkwOks5qcdrSgaJpZM4Ip8Gm>
> .
>